tactics/hints.ml

```diff
@@ -1007,8 +1007,8 @@ let make_extern pri pat tacast =
 let make_mode ref m =
   let open Term in
   let ty, _ = Typeops.type_of_global_in_context (Global.env ()) ref in
-  let ctx, t = decompose_prod ty in
-  let n = List.length ctx in
+  let ctx, t = decompose_prod_decls ty in
+  let n = Context.Rel.nhyps ctx in
```

Comment:
why this change?

Reply:
Another bugfix: let-bindings should not matter for the mode decl